extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseEntityConverter.java

```diff
@@ -4,26 +4,38 @@
 import java.util.List;
 import java.util.Map;
 
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 import org.jboss.resteasy.core.Headers;
 import org.jboss.resteasy.specimpl.BuiltResponse;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.ResponseEntity;
 
+/**
+ * This is only used in the generated ExceptionMappers when the Spring @RestControllerAdvice method returns a ResponseEntity
+ */
 public class ResponseEntityConverter {
 
-    public static Response toResponse(ResponseEntity responseEntity) {
+    public static Response toResponse(ResponseEntity responseEntity, boolean addJsonContentTypeIfNotSet) {
         return new BuiltResponse(
-                responseEntity.getStatusCodeValue(), toHeaders(responseEntity.getHeaders()), responseEntity.getBody(),
+                responseEntity.getStatusCodeValue(), toHeaders(responseEntity.getHeaders(), addJsonContentTypeIfNotSet),
+                responseEntity.getBody(),
                 new Annotation[0]);
     }
 
-    private static Headers<Object> toHeaders(HttpHeaders springHeaders) {
+    private static Headers<Object> toHeaders(HttpHeaders springHeaders, boolean addJsonContentTypeIfNotSet) {
         Headers<Object> jaxRsHeaders = new Headers<>();
         for (Map.Entry<String, List<String>> entry : springHeaders.entrySet()) {
             jaxRsHeaders.addAll(entry.getKey(), entry.getValue().toArray(new Object[0]));
         }
+        /*
+         * We add the default application/json content type if no content type is specified
+         * since this is the default value when returning an object from a Spring RestController
+         */
+        if (!jaxRsHeaders.containsKey(javax.ws.rs.core.HttpHeaders.CONTENT_TYPE) && addJsonContentTypeIfNotSet) {
```

Comment:
It's very minor as it's probably the default case but I would revert the condition to avoid the map lookup when we can.

Reply:
Makes sense

Reply:
Fixed